### PR TITLE
Fix auth check bug

### PR DESCRIPTION
**Pull request by lunny**
Fix https://github.com/go-gitea/gitea/pull/24362/files#r1179095324

`getAuthenticatedMeta` has checked them, these code are duplicated one. And the first invokation has a wrong permission check. `DownloadHandle` should require read permission but not write.

### CODE REVIEW

**Review by jolheiser — APPROVED** [review]

**Comment by lunny** [comment]
I will add some tests

**Review by 6543 — APPROVED** [review]

File: tests/integration/lfs_getobject_test.go [hunk 1]
```diff
@@ -89,6 +115,21 @@ func TestGetLFSSmall(t *testing.T) {
 	checkResponseTestContentEncoding(t, &content, resp, false)
 }
 
+func TestGetLFSSmallToken(t *testing.T) {
```
Comment (wxiaoguang) [inline tests/integration/lfs_getobject_test.go:118]: `TestGetLFSSmallToken`

What is a "small token"?

Comment (jolheiser) [reply tests/integration/lfs_getobject_test.go:118]: I think it's "Get LFS small (file with) Token"

Comment (lunny) [reply tests/integration/lfs_getobject_test.go:118]: Hm, maybe it should be `with token` because there is a similar tests with username/password

**Pull request merged**